presto-hive/src/test/java/com/facebook/presto/hive/containers/HiveMinIODataLake.java

```diff
@@ -119,25 +119,103 @@ public HiveMinIODataLake(String bucketName, Map<String, String> hiveHadoopFilesT
 
     public void start()
     {
-        if (isStarted()) {
+        if (!isStarted.compareAndSet(false, true)) {
             return;
         }
+
         try {
             this.minIOContainer.start();
+            log.info("MinIO started on port: " + minIOContainer.getMinioApiEndpoint().getPort());
+
+            waitForMinIO();
+
             this.hiveHadoopContainer.start();
-            AmazonS3 s3Client = AmazonS3ClientBuilder
-                    .standard()
-                    .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(
-                            "http://localhost:" + minIOContainer.getMinioApiEndpoint().getPort(),
-                            "us-east-1"))
-                    .withPathStyleAccessEnabled(true)
-                    .withCredentials(new AWSStaticCredentialsProvider(
-                            new BasicAWSCredentials(ACCESS_KEY, SECRET_KEY)))
+            log.info("Hive Hadoop container started");
+
+            waitForHiveMetastore();
+
+            createS3Bucket();
+        }
+        catch (Exception e) {
+            isStarted.set(false);
+            throw new RuntimeException("Failed to start data lake", e);
+        }
+    }
+
+    private void waitForMinIO() throws InterruptedException
+    {
+        int maxRetries = 30;
+        int retryDelayMs = 1000;
+
+        for (int i = 0; i < maxRetries; i++) {
+            try {
+                S3Client testClient = S3Client.builder()
+                        .endpointOverride(URI.create("http://localhost:" + minIOContainer.getMinioApiEndpoint().getPort()))
+                        .region(Region.US_EAST_1)
+                        .credentialsProvider(StaticCredentialsProvider.create(
+                                AwsBasicCredentials.create(ACCESS_KEY, SECRET_KEY)))
+                        .forcePathStyle(true)
+                        .build();
```

Comment:
Let's add a utility method for this, as the client is being created in multiple methods